ports/raspberrypi/bindings/rp2clock/AuxSrc.c

```diff
@@ -0,0 +1,179 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/runtime.h"
+#include "supervisor/shared/translate/translate.h"
+#include "bindings/rp2clock/AuxSrc.h"
+
+//| class AuxSrc:
+//|     """Defines the input clock for GPOUT on RP2040"""
+//|
+//|     def __init__(self) -> None:
+//|         """Enum-like class to define the clock src."""
+//|
+const mp_obj_type_t rp2clock_auxsrc_type;
+
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_pll_sys_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_gpin0_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_gpin1_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_pll_usb_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_pll_rosc_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_pll_xosc_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_sys_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_usb_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_adc_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_rtc_obj = {
+    { &rp2clock_auxsrc_type },
+};
+const rp2clock_auxsrc_obj_t rp2clock_auxsrc_ref_obj = {
+    { &rp2clock_auxsrc_type },
+};
+
+
+STATIC const mp_rom_map_elem_t rp2clock_auxsrc_locals_dict_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_PLL_SYS),  MP_ROM_PTR(&rp2clock_auxsrc_pll_sys_obj) },
+    { MP_ROM_QSTR(MP_QSTR_GPIN0),    MP_ROM_PTR(&rp2clock_auxsrc_gpin0_obj) },
+    { MP_ROM_QSTR(MP_QSTR_GPIN1),    MP_ROM_PTR(&rp2clock_auxsrc_gpin1_obj) },
+    { MP_ROM_QSTR(MP_QSTR_PLL_USB),  MP_ROM_PTR(&rp2clock_auxsrc_pll_usb_obj) },
+    { MP_ROM_QSTR(MP_QSTR_PLL_ROSC), MP_ROM_PTR(&rp2clock_auxsrc_pll_rosc_obj) },
+    { MP_ROM_QSTR(MP_QSTR_PLL_XOSC), MP_ROM_PTR(&rp2clock_auxsrc_pll_xosc_obj) },
+    { MP_ROM_QSTR(MP_QSTR_SYS),      MP_ROM_PTR(&rp2clock_auxsrc_sys_obj) },
+    { MP_ROM_QSTR(MP_QSTR_USB),      MP_ROM_PTR(&rp2clock_auxsrc_usb_obj) },
+    { MP_ROM_QSTR(MP_QSTR_ADC),      MP_ROM_PTR(&rp2clock_auxsrc_adc_obj) },
+    { MP_ROM_QSTR(MP_QSTR_RTC),      MP_ROM_PTR(&rp2clock_auxsrc_rtc_obj) },
+    { MP_ROM_QSTR(MP_QSTR_REF),      MP_ROM_PTR(&rp2clock_auxsrc_ref_obj) },
+};
```

Comment:
Please add documentation for each of these values. Here is an example: https://github.com/adafruit/circuitpython/blob/e64082d60052f92720031b5a17aec3732a6c4e26/shared-bindings/synthio/Math.c#L54 You should use the `MAKE_ENUM_VALUE()` macro as well.